frontend/src/plugins/impl/CodeEditorPlugin.tsx

```diff
@@ -57,6 +62,37 @@ const CodeEditorComponent = (props: CodeEditorComponentProps) => {
   const minHeight = props.minHeight ? `${props.minHeight}px` : "70px";
   const maxHeight = props.maxHeight ? `${props.maxHeight}px` : undefined;
 
+  const [localValue, setLocalValue] = useState(props.value);
+  const { onChange: setValueDebounced } = useDebounceControlledState<string>({
+    initialValue: props.value,
+    delay: Number.isFinite(props.debounce) ? (props.debounce as number) : 0,
+    onChange: props.setValue,
+    disabled: !Number.isFinite(props.debounce),
+  });
+
+  const handleChange = useCallback(
+    (newValue: string) => {
+      setLocalValue((_) => newValue);
+      if (typeof props.debounce === "number") {
+        setValueDebounced(newValue);
+      } else if (props.debounce !== true) {
+        props.setValue(newValue);
+      }
+    },
+    [setValueDebounced, props.debounce, props.setValue],
+  );
+
+  const extensions =
+    props.debounce === true
+      ? [
+          events.content({
```

Comment:
could we use: `EditorView.domEventHandlers({` blur: ()...` instead of bringing in another library?

i assume that library exists for a reason, so maybe it does more than `EditorView.domEventHandlers`? if so, can you add a note why?


also could we memoize the extensions:

```
const onBlur = useEvent(() => {
    return props.setValue(localValue);  
});

const extensions = useMemo(() => {
   if (props.debounce === true) {
        return [events.content({ blur: onBlur })]
   }

    return []
}, [props.debounce, onBlur]
```

the reason for `useEvent` is that this callback can stay memorized even if `localValue` changes.